### PR TITLE
Legacy split clone workflow

### PR DESCRIPTION
**Pull request by rafael**
### Description

This PR adds the option to use `LegacySplitClone` during an horizontal resharding workflow.  The workflow UI will know like the following:

image

### Tests

I tested this in our dev environment and it all worked as expected.

### CODE REVIEW

File: web/vtctld2/package.json [hunk 1]
```diff
@@ -53,6 +53,6 @@
     "protractor": "3.3.0",
     "ts-node": "1.2.1",
     "tslint": "3.13.0",
-    "typescript": "^2.0.0"
+    "typescript": "=2.0.10"
```
Comment (rafael) [inline web/vtctld2/package.json]: I had to pin down the `typescript` version to be able to bundle the assets using `make web_build`. However I'm not sure if this is only needed in MacOS or something else environment specific.

File: go/vt/workflow/resharding/horizontal_resharding_workflow.go [hunk 2]
```diff
@@ -76,19 +76,21 @@ func (*HorizontalReshardingWorkflowFactory) Init(m *workflow.Manager, w *workflo
 	subFlags := flag.NewFlagSet(horizontalReshardingFactoryName, flag.ContinueOnError)
 	keyspace := subFlags.String("keyspace", "", "Name of keyspace to perform horizontal resharding")
 	vtworkersStr := subFlags.String("vtworkers", "", "A comma-separated list of vtworker addresses")
+	minHealthyRdonlyTablets := subFlags.String("min_healthy_rdonly_tablets", "", "Minimum number of rdonly healthy tablets required")
```
Comment (demmer) [inline go/vt/workflow/resharding/horizontal_resharding_workflow.go]: Minor nit -- I'd rephrase to "Minimum number of healthy RDONLY tablets required"

Comment (demmer) [inline go/vt/workflow/resharding/horizontal_resharding_workflow.go]: Minor nit -- I'd rephrase to "Minimum number of healthy RDONLY tablets required"

File: go/vt/workflow/resharding/horizontal_resharding_workflow.go [hunk 3]
```diff
@@ -76,19 +76,21 @@ func (*HorizontalReshardingWorkflowFactory) Init(m *workflow.Manager, w *workflo
...
+	splitCmd := subFlags.String("split_cmd", "", "Split command to use to perform horizontal resharding (either SplitClone or LegacySplitClone)")
 	enableApprovals := subFlags.Bool("enable_approvals", true, "If true, executions of tasks require user's approvals on the UI.")
 
 	if err := subFlags.Parse(args); err != nil {
 		return err
 	}
-	if *keyspace == "" || *vtworkersStr == "" {
-		return fmt.Errorf("Keyspace name, vtworkers information must be provided for horizontal resharding")
+	if *keyspace == "" || *vtworkersStr == "" || *minHealthyRdonlyTablets == "" || *splitCmd == "" {
+		return fmt.Errorf("Keyspace name, min healthy rdonly tablets and vtworkers information must be provided for horizontal resharding")
```
Comment (demmer) [inline go/vt/workflow/resharding/horizontal_resharding_workflow.go]: nit: "Keyspace name, min healthy rdonly tablets, split command, and vtworkers "

File: go/vt/workflow/resharding/horizontal_resharding_workflow.go (hunk 2 above)
Comment (demmer) [reply go/vt/workflow/resharding/horizontal_resharding_workflow.go]: Also this should use `subFlags.Int` and should be passed through the various layers as an int, not a string.

File: go/vt/workflow/resharding/tasks.go [hunk 4]
```diff
@@ -68,17 +68,17 @@ func (hw *HorizontalReshardingWorkflow) runSplitClone(ctx context.Context, t *wo
 	keyspace := t.Attributes["keyspace"]
 	sourceShard := t.Attributes["source_shard"]
 	worker := t.Attributes["vtworker"]
+	minHealhtyRdonlyTablets := t.Attributes["min_healthy_rdonly_tablets"]
```
Comment (demmer) [inline go/vt/workflow/resharding/tasks.go]: typo: `minHealthyRdonlyTablets` is what you meant I think

File: web/vtctld2/src/app/shared/flags/workflow.flags.ts [hunk 5]
```diff
@@ -111,6 +134,13 @@ export class HorizontalReshardingVtworkerFlag extends InputFlag {
   }
 }
 
+export class HorizontalReshardingMinHealthyRdonlyTablets extends InputFlag {
+  constructor(position: number, id: string, value= '') {
+    super(position, id, 'min healthy rdonly tablets', 'Minimum number of rdonly healthy tablets required', value);
```
Comment (demmer) [inline web/vtctld2/src/app/shared/flags/workflow.flags.ts]: Same comment as above re this string.

File: go/vt/workflow/resharding/horizontal_resharding_workflow.go (hunk 2 above)
Comment (rafael) [reply go/vt/workflow/resharding/horizontal_resharding_workflow.go]: I did as strings because they all get passed down this way:

```
args := []string{splitCmd, "--min_healthy_rdonly_tablets", minHealthyRdonlyTablets, sourceKeyspaceShard}
```

If we do it as an int I have to convert back to string in here. Do you is worth it?

**Comment by rafael** [comment]
Most of the comments have been addressed. There was also an issue with the tests. I think it should be fixed, but let's wait for travis.

File: go/vt/workflow/resharding/horizontal_resharding_workflow.go (hunk 2 above)
Comment (demmer) [reply go/vt/workflow/resharding/horizontal_resharding_workflow.go]: Ah yeah -- sorry I noticed this later and forgot to delete my earlier comment. I think it's fine as-is.

Ideally we would want some validation in the UI layer to restrict this to only accept numbers since I think otherwise we'll only find out when the split clone process fails, but that's a pretty minor issue.
